java/source/dataset.rst

```diff
@@ -25,6 +25,24 @@ Dataset
 
 .. contents::
 
+Arrow Java Dataset offer native functionalities consuming native artifacts such as:
+    - JNI Arrow C++ Dataset: libarrow_dataset_jni (dylib/so):
+        To create C++ natively objects Schema, Dataset, Scanner and export that as a references (long id).
+    - JNI Arrow C Data Interface: libarrow_cdata_jni (dylib/so):
+        To get C++ Recordbacth.
```

Comment:
I don't think we want to talk about implementation details here

Reply:
Deleted